standalone/include/psen_scan_v2_standalone/data_conversion_layer/laserscan_conversions.h

```diff
@@ -79,59 +84,76 @@ LaserScanConverter::toLaserScan(const std::vector<data_conversion_layer::monitor
   scan.setMeasurements(measurements);
   scan.setIntensities(intensities);
 
   return scan;
 }
 
 inline std::vector<int> LaserScanConverter::getFilledFramesIndicesSortedByThetaAngle(
-    const std::vector<data_conversion_layer::monitoring_frame::Message>& frames)
+    const std::vector<data_conversion_layer::monitoring_frame::MessageStamped>& frames)
 {
   std::vector<int> sorted_filled_frames_indices(frames.size());
   std::iota(sorted_filled_frames_indices.begin(), sorted_filled_frames_indices.end(), 0);
   std::sort(sorted_filled_frames_indices.begin(), sorted_filled_frames_indices.end(), [frames](int i1, int i2) {
-    return frames[i1].fromTheta() < frames[i2].fromTheta();
+    return frames[i1].msg_.fromTheta() < frames[i2].msg_.fromTheta();
   });
 
   // The following contains a missing line in the coverage report, which does not make sense.
   // LCOV_EXCL_START
   sorted_filled_frames_indices.erase(std::remove_if(sorted_filled_frames_indices.begin(),
                                                     sorted_filled_frames_indices.end(),
-                                                    [frames](int i) { return frames[i].measurements().empty(); }),
+                                                    [frames](int i) { return frames[i].msg_.measurements().empty(); }),
                                      sorted_filled_frames_indices.end());
   // LCOV_EXCL_STOP
 
   return sorted_filled_frames_indices;
 }
 
-inline util::TenthOfDegree
-LaserScanConverter::calculateMaxAngle(const std::vector<data_conversion_layer::monitoring_frame::Message>& frames,
-                                      const util::TenthOfDegree& min_angle)
+inline util::TenthOfDegree LaserScanConverter::calculateMaxAngle(
+    const std::vector<data_conversion_layer::monitoring_frame::MessageStamped>& frames,
+    const util::TenthOfDegree& min_angle)
 {
-  const auto resolution = frames[0].resolution();
+  const auto resolution = frames[0].msg_.resolution();
   const uint16_t number_of_samples =
       std::accumulate(frames.begin(), frames.end(), uint16_t{ 0 }, [](uint16_t total, const auto& frame) {
-        return total + frame.measurements().size();
+        return total + frame.msg_.measurements().size();
       });
   return min_angle + resolution * static_cast<int>(number_of_samples - 1);
 }
 
+inline int64_t LaserScanConverter::calculateTimestamp(
+    const std::vector<data_conversion_layer::monitoring_frame::MessageStamped>& frames,
+    const std::vector<int>& filled_frames_indices)
+{
+  const auto it = std::min_element(filled_frames_indices.begin(), filled_frames_indices.end(), [&frames](int i, int j) {
+    return frames[i].stamp_ < frames[j].stamp_;
+  });
+  const auto& frame{ frames[*it] };
+  return frame.stamp_ -
+         static_cast<int64_t>(std::round(frame.msg_.resolution().value() * (frame.msg_.measurements().size() - 1) *
+                                         configuration::TIME_PER_SCAN_IN_S * 1000000000.0 / 3600.0));
```

Comment:
Kann man hier durch auslagern in andere Methoden oder notfalls Comments noch bisle Klarheit über den Vorgang herstellen?
get_first_frame()
calculate_time_of_first_ray() <- vorallem das 1000000000/3600 ist denke ich nicht direkt ersichtlich

Reply:
Better now?